src/map.js

```diff
@@ -649,7 +647,30 @@ var map = function (arg) {
         height: m_height
       };
     }
-    m_this.resize(0, 0, arg.width, arg.height);
+    // store the original center and restore it after the resize
+    var oldCenter = m_this.center();
```

Comment:
I don't think this needs to be in an internal method.  It is specific to resizing, and that can all go through the `size` method.